java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/IndexController.java

```diff
@@ -138,4 +205,49 @@ public String listPhp(@RequestParam(required = false) String sortby,
         model.addAttribute("asc", asc);
         return "pub/list";
     }
+
+    @GetMapping(path = {"check_live", "check_live.php"}, produces = MediaType.APPLICATION_XML_VALUE)
+    @ResponseBody
+    public ResponseEntity<String> checkLive(@RequestParam("file") String file,
+                                            @RequestParam("timestamp") Long timestamp) {
+        if (file.isEmpty() || timestamp == null) {
+            return ResponseEntity.badRequest().body("BadRequest");
+        }
+        XMLOutputFactory output = XMLOutputFactory.newInstance();
+        XMLStreamWriter xmlWriter = null;
+        StringWriter stringWriter = null;
+        try {
+            stringWriter = new StringWriter();
+            xmlWriter = output.createXMLStreamWriter(stringWriter);
+            xmlWriter.writeStartDocument();
+            xmlWriter.writeStartElement("updates");
+            xmlWriter.writeAttribute("file", file);
+            xmlWriter.writeAttribute("timestamp", String.valueOf(timestamp));
+            String folder = "aosmc".concat(File.separator).concat(file.toLowerCase()).concat(File.separator);
+            File dir = new File(filesLocation, folder);
+            if (dir.isDirectory()) {
+                for (File mapFile : dir.listFiles()) {
+                    String filename = mapFile.getName().toLowerCase();
+                    if (filename.startsWith(file)) {
+                        writeAttributes(xmlWriter, mapFile, timestamp);
+                    }
+                }
+            }
+            xmlWriter.writeEndElement();
+            xmlWriter.writeEndDocument();
+            xmlWriter.flush();
+        } catch (XMLStreamException ex) {
+            LOGGER.error(ex.getMessage(), ex);
```

Comment:
no need for stack trace - the log is huge with it. LOGGER.error(" Error within checklive: "+ ex.getMessage());